### PR TITLE
Dịch "rnn_vn" - Phần 4

### LINKED ISSUE #1507: Dịch "rnn_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recurrent-neural-networks/rnn_vn.md#L202-L278

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tranyenthy**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1507 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 1]
```diff
@@ -273,7 +288,13 @@ Nonetheless, at least the model has learned how to spell words and some degree o
 Last, example 3 indicates a poorly trained model that does not fit data properly.
 -->
 
-*dịch đoạn phía trên*
+Về chất lượng, ví dụ 1 rõ ràng là tốt nhất.
+Các từ là hợp lý và hợp lý mạch lạc.
+Mặc dù nó có thể không hoàn toàn phản ánh chính xác từ nào theo ngữ nghĩa ("ở San Francisco" và "vào mùa đông" sẽ là phần mở rộng hoàn toàn hợp lý),
+mô hình có thể nắm bắt loại từ nào theo sau.
+Ví dụ 2 tệ hơn đáng kể bằng cách tạo ra một phần mở rộng vô nghĩa.
+Tuy nhiên, ít nhất mô hình đã học cách đánh vần các từ và một số mức độ tương quan giữa các từ.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/rnn_vn.md:296]:
```suggestion
Tuy nhiên, ít nhất mô hình đã viết đúng các từ và học được một số mức độ tương quan giữa các từ.
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 2]
```diff
@@ -273,7 +288,13 @@ Nonetheless, at least the model has learned how to spell words and some degree o
...
+Cuối cùng, ví dụ 3 chỉ ra một mô hình được huấn luyện kém, không phù hợp với dữ liệu.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/rnn_vn.md:297]:
```suggestion
Cuối cùng, ví dụ 3 là một mô hình được huấn luyện kém, không phù hợp với dữ liệu.
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 3]
```diff
@@ -244,7 +254,7 @@ Therefore, the input $\mathbf X_t$ at timestep $t$ will be a $n\times d$ matrix,
 ## Perplexity
 -->
 
-## *dịch tiêu đề phía trên*
+## Độ Bối rối
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/rnn_vn.md:257]: #1598

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 4]
```diff
@@ -253,15 +263,20 @@ A good language model is able to predict with high accuracy tokens that what we
 Consider the following continuations of the phrase "It is raining", as proposed by different language models:
 -->
 
-*dịch đoạn phía trên*
+Cuối cùng, chúng ta hãy thảo luận về cách đo lường chất lượng mô hình chuỗi.
+Một cách là kiểm tra tính gây ngạc nhiên của văn bản.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/rnn_vn.md:267]:
```suggestion
Một cách làm là ta kiểm tra độ ngạc nhiên của văn bản.
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 5]
```diff
@@ -273,7 +288,13 @@ Nonetheless, at least the model has learned how to spell words and some degree o
 Last, example 3 indicates a poorly trained model that does not fit data properly.
 -->
 
-*dịch đoạn phía trên*
+Về chất lượng, ví dụ 1 rõ ràng là tốt nhất.
+Các từ là hợp lý và hợp lý mạch lạc.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/rnn_vn.md:292]:
```suggestion
Các từ là hợp lý và mạch lạc về mặt logic.
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 6]
```diff
@@ -220,21 +220,31 @@ Since the next word of the sequence in the training data is "by", the loss of ti
 the probability distribution of the next word generated based on the feature sequence "the", "time", "machine" and the label "by" of this timestep.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta minh họa cách RNN có thể được sử dụng để xây dựng mô hình ngôn ngữ.
+Để đơn giản, chúng tôi sử dụng các từ thay vì các ký tự làm đầu vào, vì từ dễ hiểu hơn.
+Đặt kích thước minibatch là 1, chuỗi văn bản là phần đầu của tập dữ liệu, "the time machine by H. G. Wells".
+:numref:`fig_rnn_train` minh họa cách ước đoán từ tiếp theo dựa trên các từ hiện tại và trước đó.
+Trong quá trình huấn luyện, chúng ta áp dụng softmax vào đầu ra tại mỗi bước thời gian,
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/rnn_vn.md:227]:
```suggestion
Trong quá trình huấn luyện, chúng ta áp dụng softmax cho đầu ra tại mỗi bước thời gian,
```

File: chapter_recurrent-neural-networks/rnn_vn.md (hunk 3 above)
Comment (rootonchair) [inline chapter_recurrent-neural-networks/rnn_vn.md:257]:
```suggestion
## Độ Rối rắm
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 7]
```diff
@@ -273,7 +288,13 @@ Nonetheless, at least the model has learned how to spell words and some degree o
 Last, example 3 indicates a poorly trained model that does not fit data properly.
 -->
 
-*dịch đoạn phía trên*
+Về chất lượng, ví dụ 1 rõ ràng là tốt nhất.
+Các từ là hợp lý và mạch lạc về mặt logic.
+Mặc dù nó có thể không hoàn toàn phản ánh chính xác từ nào theo ngữ nghĩa ("ở San Francisco" và "vào mùa đông" sẽ là phần mở rộng hoàn toàn hợp lý),
+mô hình có thể nắm bắt loại từ nào theo sau.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/rnn_vn.md:294]:
```suggestion
mô hình vẫn có thể nắm bắt loại từ nào nên theo sau.
```

File: chapter_recurrent-neural-networks/rnn_vn.md [hunk 8]
```diff
@@ -273,7 +288,13 @@ Nonetheless, at least the model has learned how to spell words and some degree o
 Last, example 3 indicates a poorly trained model that does not fit data properly.
 -->
 
-*dịch đoạn phía trên*
+Về chất lượng, ví dụ 1 rõ ràng là tốt nhất.
+Các từ là hợp lý và mạch lạc về mặt logic.
+Mặc dù nó có thể không hoàn toàn phản ánh chính xác từ nào theo ngữ nghĩa ("ở San Francisco" và "vào mùa đông" sẽ là phần mở rộng hoàn toàn hợp lý),
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/rnn_vn.md:293]:
```suggestion
Mặc dù nó có thể không phản ánh từ nào nên theo sau một cách chính xác về mặt ngữ nghĩa  ("ở San Francisco" và "vào mùa đông" sẽ là phần mở rộng hợp lý hơn),
```
